cloudinit/sources/__init__.py

```diff
@@ -37,36 +47,45 @@
 from cloudinit.persistence import CloudInitPickleMixin
 from cloudinit.reporting import events
 
-DSMODE_DISABLED = "disabled"
-DSMODE_LOCAL = "local"
-DSMODE_NETWORK = "net"
-DSMODE_PASS = "pass"
+DsMode = Literal["disabled", "local", "net", "pass"]
+DSMODE_DISABLED: Final[DsMode] = "disabled"
+DSMODE_LOCAL: Final[DsMode] = "local"
+DSMODE_NETWORK: Final[DsMode] = "net"
+DSMODE_PASS: Final[DsMode] = "pass"
 
 VALID_DSMODES = [DSMODE_DISABLED, DSMODE_LOCAL, DSMODE_NETWORK]
 
-DEP_FILESYSTEM = "FILESYSTEM"
-DEP_NETWORK = "NETWORK"
-DS_PREFIX = "DataSource"
+# DEP_FILESYSTEM and DEP_NETWORK are part of an abstraction that was never
+# fully realized and has unnecessary complexity.
+#
+# [DEP_FILESYSTEM]              - this means Local stage
+# [DEP_FILESYSTEM, DEP_NETWORK] - this means Network stage
+#
+# XXX: TODO - simplify this logic
+Deps = Literal["FILESYSTEM", "NETWORK"]
+DEP_FILESYSTEM: Deps = "FILESYSTEM"
+DEP_NETWORK: Deps = "NETWORK"
+DS_PREFIX: Final = "DataSource"
 
-EXPERIMENTAL_TEXT = (
+EXPERIMENTAL_TEXT: Final = (
     "EXPERIMENTAL: The structure and format of content scoped under the 'ds'"
     " key may change in subsequent releases of cloud-init."
 )
 
 
-REDACT_SENSITIVE_VALUE = "redacted for non-root user"
+REDACT_SENSITIVE_VALUE: Final = "redacted for non-root user"
 
 # Key which can be provide a cloud's official product name to cloud-init
-METADATA_CLOUD_NAME_KEY = "cloud-name"
+METADATA_CLOUD_NAME_KEY: Final = "cloud-name"
 
-UNSET = "_unset"
-METADATA_UNKNOWN = "unknown"
+UNSET: Final[dict] = {}
```

Comment:
This variable is referenced in a bunch of places outside of this file. I think I'd rather see us replace instances of this one by one rather than trying to do a big bang update here.